src/io_threads.c

```diff
@@ -375,3 +409,149 @@ int trySendWriteToIOThreads(client *c) {
     IOJobQueue_push(jq, ioThreadWriteToClient, c);
     return C_OK;
 }
+
+/* Internal function to free the client's argv in an IO thread. */
+void IOThreadFreeArgv(void *data) {
+    robj **argv = (robj **)data;
+
+    for (int i = 0;; i++) {
+        robj *o = argv[i];
+        if (o == NULL) {
+            continue;
+        }
+
+        int allocator_id = o->allocator_id;
+        decrRefCount(o);
+
+        /* The allocator_id is set to 0 to indicate that this is the last argument to free */
+        if (allocator_id == 0) {
+            break;
+        }
+    }
+
+    zfree(argv);
+}
+
+/* This function attempts to offload the client's argv to an IO thread.
+ * Returns C_OK if the client's argv were successfully offloaded to an IO thread,
+ * C_ERR otherwise. */
+int tryOffloadFreeArgvToIOThreads(client *c) {
```

Comment:
We only try to offload it to one thread, right?

```suggestion
int tryOffloadFreeArgvToIOThread(client *c) {
```

Reply:
We always send a job to only one thread.
All the ioThreads API functions are named with 'threads' in the plural form. I'm not sure if we need to change this naming convention